### ISSUE TITLE
Improve accessibility with pa11y

### ISSUE THREAD

**Issue by alexsoble** [codeforamerica/somerville-teacher-tool#130]
[pa11y](https://github.com/nature/pa11y) is an automated accessibility testing pal. 

Using pa11y will help us make sure our web app is accessible.

**Comment by alexsoble** [comment]
Here's are pa11y tips for our homepage:

# Welcome to Pa11y

## Results for https://somerville-teacher-tool.herokuapp.com/:
* __Notice:__ Check that the title element describes the document.
 * WCAG2AA.Principle2.Guideline2_4.2_4_2.H25.2
 * html > head > title
* __Error:__ The html element should have a lang or xml:lang attribute which describes the language of the document.
 * WCAG2AA.Principle3.Guideline3_1.3_1_1.H57.2
 * html
* __Error:__ Heading tag found with no content. Text that is not intended as a heading should not be marked up with heading tags.
 * WCAG2AA.Principle1.Guideline1_3.1_3_1.H42.2
 * #title
* __Error:__ This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.29:1. Recommendation: change text colour to #3177c9.
 * WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
 * html > body > div:nth-child(1) > div > nav > a:nth-child(1)
* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(1) > div > nav > a:nth-child(1)
* __Error:__ This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.29:1. Recommendation: change text colour to #3177c9.
 * WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
 * html > body > div:nth-child(1) > div > nav > a:nth-child(2)
* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(1) > div > nav > a:nth-child(2)
* __Warning:__ The heading structure is not logically nested. This h5 element should be an h2 to be properly nested.
 * WCAG2AA.Principle1.Guideline1_3.1_3_1_A.G141
 * html > body > div:nth-child(2) > div > div:nth-child(1) > h5
* __Error:__ This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.29:1. Recommendation: change text colour to #3177c9.
 * WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
 * html > body > div:nth-child(2) > div > div:nth-child(1) > p > a
* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(2) > div > div:nth-child(1) > p > a
* __Error:__ This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.29:1. Recommendation: change text colour to #3177c9.
 * WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
 * html > body > div:nth-child(2) > div > div:nth-child(3) > p:nth-child(4) > a
* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(2) > div > div:nth-child(3) > p:nth-child(4) > a
* __Warning:__ If this element contains a navigation section, it is recommended that it be marked up as a list.
 * WCAG2AA.Principle1.Guideline1_3.1_3_1.H48
 * html > body > div:nth-child(2) > div > div:nth-child(4)
* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(1)
* __Notice:__ Ensure that the img element's alt text serves the same purpose and presents the same information as the image.
 * WCAG2AA.Principle1.Guideline1_1.1_1_1.G94.Image
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(1) > img
* __Notice:__ If this image cannot be fully described in a short text alternative, ensure a long text alternative is also available, such as in the body text or through a link.
 * WCAG2AA.Principle1.Guideline1_1.1_1_1.G73,G74
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(1) > img
* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(2)
* __Notice:__ Ensure that the img element's alt text serves the same purpose and presents the same information as the image.
 * WCAG2AA.Principle1.Guideline1_1.1_1_1.G94.Image
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(2) > img
* __Notice:__ If this image cannot be fully described in a short text alternative, ensure a long text alternative is also available, such as in the body text or through a link.
 * WCAG2AA.Principle1.Guideline1_1.1_1_1.G73,G74
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(2) > img

## Summary:
* 6 Errors
* 2 Warnings
* 11 Notices

**Comment by alexsoble** [comment]
New report with more detail and one less error:

# Welcome to Pa11y

## Results for https://somerville-teacher-tool.herokuapp.com/:
* __Notice:__ Check that the title element describes the document.
 * WCAG2AA.Principle2.Guideline2_4.2_4_2.H25.2
 * html > head > title
 * `<title>Student Insights</title>`

* __Error:__ Heading tag found with no content. Text that is not intended as a heading should not be marked up with heading tags.
 * WCAG2AA.Principle1.Guideline1_3.1_3_1.H42.2
 * #title
 * `<h1 id="title" alt="Student Insights"></h1>`

* __Error:__ This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.29:1. Recommendation: change text colour to #3177c9.
 * WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
 * html > body > div:nth-child(1) > div > nav > a:nth-child(1)
 * `<a href="https://somerville-teacher-tool.herokuapp.com/about">About</a>`

* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(1) > div > nav > a:nth-child(1)
 * `<a href="https://somerville-teacher-tool.herokuapp.com/about">About</a>`

* __Error:__ This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.29:1. Recommendation: change text colour to #3177c9.
 * WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
 * html > body > div:nth-child(1) > div > nav > a:nth-child(2)
 * `<a href="/educators/sign_in">Sign In</a>`

* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(1) > div > nav > a:nth-child(2)
 * `<a href="/educators/sign_in">Sign In</a>`

* __Warning:__ The heading structure is not logically nested. This h5 element should be an h2 to be properly nested.
 * WCAG2AA.Principle1.Guideline1_3.1_3_1_A.G141
 * html > body > div:nth-child(2) > div > div:nth-child(1) > h5
 * `<h5>Who is it for</h5>`

* __Error:__ This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.29:1. Recommendation: change text colour to #3177c9.
 * WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
 * html > body > div:nth-child(2) > div > div:nth-child(1) > p > a
 * `<a href="http://www.codeforamerica.org/governments/somerville/" target="_blank">Team Somerville</a>`

* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(2) > div > div:nth-child(1) > p > a
 * `<a href="http://www.codeforamerica.org/governments/somerville/" target="_blank">Team Somerville</a>`

* __Error:__ This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 3.29:1. Recommendation: change text colour to #3177c9.
 * WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
 * html > body > div:nth-child(2) > div > div:nth-child(3) > p:nth-child(4) > a
 * `<a href="https://github.com/codeforamerica/somerville-teacher-tool" target="_blank">GitHub.</a>`

* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(2) > div > div:nth-child(3) > p:nth-child(4) > a
 * `<a href="https://github.com/codeforamerica/somerville-teacher-tool" target="_blank">GitHub.</a>`

* __Warning:__ If this element contains a navigation section, it is recommended that it be marked up as a list.
 * WCAG2AA.Principle1.Guideline1_3.1_3_1.H48
 * html > body > div:nth-child(2) > div > div:nth-child(4)
 * `<div class="logos"> <a href="http://www.some...</div>`

* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(1)
 * `<a href="http://www.somervillema.gov/" target="_blank" class="somerville-logo"><img alt="Somerville logo" src=...</a>`

* __Notice:__ Ensure that the img element's alt text serves the same purpose and presents the same information as the image.
 * WCAG2AA.Principle1.Guideline1_1.1_1_1.G94.Image
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(1) > img
 * `<img alt="Somerville logo" src="/assets/somerville_logo-1490ad234c5164dcdcce25103217e3d6.png">`

* __Notice:__ If this image cannot be fully described in a short text alternative, ensure a long text alternative is also available, such as in the body text or through a link.
 * WCAG2AA.Principle1.Guideline1_1.1_1_1.G73,G74
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(1) > img
 * `<img alt="Somerville logo" src="/assets/somerville_logo-1490ad234c5164dcdcce25103217e3d6.png">`

* __Notice:__ Check that the link text combined with programmatically determined link context identifies the purpose of the link.
 * WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(2)
 * `<a href="https://www.codeforamerica.org/" target="_blank" class="cfa-logo"><img alt="Cfa logo" src="/asset...</a>`

* __Notice:__ Ensure that the img element's alt text serves the same purpose and presents the same information as the image.
 * WCAG2AA.Principle1.Guideline1_1.1_1_1.G94.Image
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(2) > img
 * `<img alt="Cfa logo" src="/assets/cfa_logo-f993394e431b2439ea6e315346c37811.png">`

* __Notice:__ If this image cannot be fully described in a short text alternative, ensure a long text alternative is also available, such as in the body text or through a link.
 * WCAG2AA.Principle1.Guideline1_1.1_1_1.G73,G74
 * html > body > div:nth-child(2) > div > div:nth-child(4) > a:nth-child(2) > img
 * `<img alt="Cfa logo" src="/assets/cfa_logo-f993394e431b2439ea6e315346c37811.png">`


## Summary:
* 5 Errors
* 2 Warnings
* 11 Notices

**Comment by alexsoble** [comment]
We've taken initial steps in this direction with pa11y above, open to contributions.

**Comment by alexsoble** [comment]
We're not going to work on this in the next project milestones.
